src/class4Strings/StringAPI.java

```diff
@@ -0,0 +1,167 @@
+package class4Strings;
+
+import java.util.Scanner;
+
+public class StringAPI {
+    //GET ANY STRING VALUE
+    static String getString() {
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Please enter any String value");
+        String inputString = scanner.next();
+        System.out.println("String : " + inputString);
+        return inputString;
+    }
+
+    //GET ANY INT VALUE
+    static int getInt() {
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Please enter any integer value");
+        while (!scanner.hasNextInt()) {
+            //Error message when entering non integer value
+            System.out.println("Please enter only integer value");
+            scanner.next();
+        }
+        int inputInt = scanner.nextInt();
+        //Error message when entering negative numeric value or zero
+        while (inputInt <= 0) {
+            System.out.println("Please enter only positive integer value");
+            //Error message when entering non integer value
+            while (!scanner.hasNextInt()) {
+                System.out.println("Please enter only integer value");
+                scanner.next();
+            }
+            inputInt = scanner.nextInt();
+        }
+        System.out.println("Int : " + inputInt);
+        return inputInt;
+    }
+
+    //FIND THE LETTER OF STRING WITH INPUT INDEX
+    static void findIndexLetter(String getString, int getInt) {
+        //Create array of char type
+        char[] ch = new char[getString.length()];
+        //Fill-in String elements in char array
+        for (int i = 0; i < getString.length(); ++i) {
+            ch[i] = getString.charAt(i);
+            if (i == getInt) {
+                System.out.println("The character at index " + i + " of your String is letter " + ch[i - 1]);
+                return;
+            }
+        }
+        System.out.println("Out of bounds");
+    }
+
+    //FIND NUMBER OF WORDS IN A TEXT
+    static void findNumberOfWords(String getString) {
+        int count = 0;
+        String[] startWord = getString.split(" ");
+        for (int i = 0; i < startWord.length; ++i) {
+            if (!startWord[i].isEmpty()) {
+                count++;
+            }
+        }
+        System.out.println("Number of words : " + (count));
+    }
+
+    //CHECK IF STRING IS PALINDROME
+    static void checkIfPalindrome(String getString) {
```

Comment:
if (yourString.equals(new StringBuilder(yourString).reverse())) --> yes, it's polidrome!

:)

Reply:
Thank you for comments.
Just made all the suggested changes and merged the PR.